webapp/components/pokemon-table.component.ts

```diff
@@ -20,13 +20,41 @@ export class PokemonTableComponent {
 	//right now we're only allowing 1 transfer at a time, like God intended,
 	//this may need to be updated later to allow for batch transfers
 	private _transferringPokemonAtIndex: number = null;
+	private _renamingPokemonAtIndex: number = null;
 	private _currentSortOrderName: string = '';
 
-	constructor(private _properties: PropertiesService, private _pokemonService: PokemonService) { 
+	constructor(private _properties: PropertiesService, private _pokemonService: PokemonService) {
 	}
 
-	public set pokemon(pokemon: Pokemon[]){
-		this._pokemon = pokemon;
+	public set pokemon(pokemons: Pokemon[]){
+		this._pokemon = pokemons;
+
+		this._pokemon = this._pokemon.map(function (pokemon) {
+			pokemon.attacks = {
+			  fast: window['pokemon'][pokemon.pokedex_number].QuickMoves.map(function (moveNumber) {
+					var move = window['moves'][moveNumber];
+					return {
+					  type: move.Type.toLowerCase(),
```

Comment:
Sure, can you give me some insight on how to do it ? I'm not used to this design pattern

Something like this ?
```js
export class PokemonMoves {
	constructor(
		public fast: any,
		public charged: any
	){ }
}
```
Or something more detailed ? How can I describe nested properties ?